lib/commons/math/split-rects.js

```diff
@@ -37,19 +37,33 @@ function splitRect(inputRect, clipRect) {
     rects.push({ top, left, bottom, right: clipRect.left });
   }
   if (rects.length === 0) {
+    // Fully overlapping
+    if (isEnclosedRect(inputRect, clipRect)) {
+      return [];
+    }
+
     rects.push(inputRect); // No intersection
   }
+
   return rects.map(computeRect); // add x / y / width / height
 }
 
 const between = (num, min, max) => num > min && num < max;
 
 function computeRect(baseRect) {
-  return {
-    ...baseRect,
-    x: baseRect.left,
-    y: baseRect.top,
-    height: baseRect.bottom - baseRect.top,
-    width: baseRect.right - baseRect.left
-  };
+  return new window.DOMRect(
```

Comment:
I though it was weird that split rects works on DOMRects but then doesn't return one, which meant that doing `splitRect(rect1, rect2).left` would return `undefined` instead of the `x` value.